src/app/space/create/deployments/apps/delete-deployment-modal.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, ElementRef, EventEmitter, Input, OnInit, Output, ViewChild, ViewEncapsulation } from '@angular/core';
+
+import { NotificationType } from 'ngx-base';
+import { ModalDirective } from 'ngx-bootstrap/modal';
+import { Subscription } from 'rxjs';
+import { Environment } from '../models/environment';
+
+import { DeploymentsService } from '../services/deployments.service';
+import { DeploymentCardComponent } from './deployment-card.component';
+
+@Component({
+  host: {
+    'class': 'create-dialog'
+  },
+  encapsulation: ViewEncapsulation.None,
+  selector: 'delete-deployment-modal',
+  templateUrl: './delete-deployment-modal.component.html',
+  styleUrls: ['./delete-deployment-modal.component.less']
+})
+export class DeleteDeploymentModal {
+
+  @Input() host: ModalDirective;
+  @Input() applicationList: string[];
+
+  deploymentCardComponent: DeploymentCardComponent;
+  applicationId: string = '';
+  environmentName: string = '';
+
+  constructor(
+    private deploymentsService: DeploymentsService) {
+  }
+
+  public openModal(deploymentCardComponent: DeploymentCardComponent): void {
+    this.deploymentCardComponent = deploymentCardComponent;
+    this.applicationId = deploymentCardComponent.applicationId;
+    this.environmentName = deploymentCardComponent.environment.name;
+    this.host.show();
+  }
+
+  public closeModal(): void {
+    this.deploymentCardComponent = null;
+    this.applicationId = '';
+    this.environmentName = '';
+    this.host.hide();
+  }
+
+  public confirmDeletion() {
+    let success = this.deploymentCardComponent.delete();
+
+    if (success) {
+      this.applicationList.filter(id => id !== this.applicationId);
```

Comment:
The modal is the one that removes the deleted application from its parent's applications list? This seems weird. I think the modal should just present the confirmation and report what selection was made to its parent. Then the parent can decide to call its own deletion function. I also think it's the wrong path to head down to have the component manually filter out its own changes this way - this should be handled by the DeploymentsService. After a deletion request completes the service should fire off a new poll cycle, which will automatically push state changes to all subscribers.